src/plugins/data_view_field_editor/public/components/field_editor/composite_editor.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { EuiNotificationBadge, EuiFlexGroup, EuiFlexItem, EuiSpacer, EuiText } from '@elastic/eui';
+import { ScriptField } from './form_fields';
+import { useFieldEditorContext } from '../field_editor_context';
+import { useFieldPreviewContext } from '../preview';
+import { UseArray, UseField, TextField } from '../../shared_imports';
+import { TypeField } from './form_fields';
+import { RUNTIME_FIELD_OPTIONS_PRIMITIVE } from './constants';
+
+export const CompositeEditor = ({}) => {
+  const { links, existingConcreteFields } = useFieldEditorContext();
+  const { fields } = useFieldPreviewContext();
+  // const { fields } = useFieldPreviewContext();
+  // const names = fields.map((field) => field.key);
+  // console.log('fields from context', fields);
+  // what I want to do -
+  // 1. get list of fields
+  // 2. get types from `fields` in context
+
+  return (
+    <div>
+      <ScriptField existingConcreteFields={existingConcreteFields} links={links} />
+      <EuiSpacer size="xl" />
+      <UseArray path="fields">
```

Comment:
I see a few problem with your implementation. Have a look at the multiple example of `<UseArray /> ` in the stories.

You can launch them with

```
yarn storybook es_ui_shared
```